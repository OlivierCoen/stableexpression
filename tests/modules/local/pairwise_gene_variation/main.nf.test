nextflow_process {

    name "Test Process PAIRWISE_GENE_VARIATION"
    script "modules/local/pairwise_gene_variation/main.nf"
    process "PAIRWISE_GENE_VARIATION"
    tag "pairwisegenevariation"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/data/merge_counts/output/all_counts.parquet', checkIfExists: true)
                input[0] = ch_counts
                """
            }
        }

        then {
            println(process.out)
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
