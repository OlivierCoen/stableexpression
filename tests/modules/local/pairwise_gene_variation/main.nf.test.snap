{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "m_measures.csv:md5,d7916ba0dc2800fec0aa36fe99c6f743"
                ],
                "1": [
                    [
                        "PAIRWISE_GENE_VARIATION",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "PAIRWISE_GENE_VARIATION",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "3": [
                    [
                        "PAIRWISE_GENE_VARIATION",
                        "psutil",
                        "6.1.1"
                    ]
                ],
                "m_measures": [
                    "m_measures.csv:md5,d7916ba0dc2800fec0aa36fe99c6f743"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-14T18:15:52.799105117"
    }
}